examples/examples.rs

```diff
@@ -141,21 +128,19 @@ fn examples() -> Result<()> {
 
     impl UserData for Vec2 {
         fn add_methods(methods: &mut UserDataMethods<Self>) {
-            methods.add_method("magnitude", |lua, vec, _| {
+            methods.add_method("magnitude", |_, vec, _: ()| {
                 let mag_squared = vec.0 * vec.0 + vec.1 * vec.1;
-                lua.pack(mag_squared.sqrt())
+                Ok(mag_squared.sqrt())
             });
 
-            methods.add_meta_function(MetaMethod::Add, |lua, params| {
-                let hlist_pat![vec1, vec2] = lua.unpack::<HList![Vec2, Vec2]>(params)?;
-                lua.pack(Vec2(vec1.0 + vec2.0, vec1.1 + vec2.1))
+            methods.add_meta_function(MetaMethod::Add, |_, (vec1, vec2): (Vec2, Vec2)| {
+                Ok(Vec2(vec1.0 + vec2.0, vec1.1 + vec2.1))
             });
         }
     }
 
-    let vec2_constructor = lua.create_function(|lua, args| {
-        let hlist_pat![x, y] = lua.unpack::<HList![f32, f32]>(args)?;
-        lua.pack(Vec2(x, y))
+    let vec2_constructor = lua.create_function(|_, (x, y): (f32, f32)| {
+        Ok(Vec2(x, y))
```

Comment:
Yes, yes, this is definitely the way to go. `unpack`/`pack` and the HList macros add quite a bit of noise, this is very readable.